src/cf.data.pre

```diff
@@ -5639,6 +5639,13 @@ DOC_START
 	that exists, Squid will chdir() to that directory at startup
 	and coredump files will be left there.
 
+	In addition to changing the directory, the process permissions are updated
+	to enable process tracing and/or coredump file generation. The details are
+	OS-specific, but look for prctl(2) PR_SET_DUMPABLE and procctl(2)
+	PROC_TRACE_CTL documentation as guiding examples. If Squid does not know
+	how to update relevant process permissions in your build environment, it
+	logs an ENOSYS-based ERROR message.
```

Comment:
It feels wrong to log an ERROR message when Squid did not fail to update permissions. In the current `return ENOSYS` case, Squid does not even know whether such an update is necessary in the current environment! Should not that be a WARNING (or just a "for your information" status/progress message) instead?

Reply:
I find a ERROR a bit "overreacting" indeed, this is more a icing on a cake.

Reply:
Glad we agree! I will wait for @yadij opinion before proceeding with the final(?) adjustments.

Reply:
In earlier review I assumed there was some importance you knew of to bump the message up to level DBG_IMPORTANT and label as ERROR.

If this is just a nice-to-have behaviour leaving it as level-2 with a WARNING should be sufficiently high. I do not have a preference between WARNING or NOTICE.

Reply:
> In earlier review I assumed there was some importance you knew of to bump the message up to level DBG_IMPORTANT and label as ERROR.

Your assumption was correct. In fact, I explicitly [stated](https://github.com/squid-cache/squid/pull/974#discussion_r800069683) the exact reason -- we should let the admin know (that there was a failure in applying their configuration directive). I do not consider level-2+ messages as "letting admin know". Level-2+ messages can be essential in triage, but they do not work well as admin notifications. Level-2+ is "pull", level-0/1 can be considered "push". Configuration failures deserve push notifications.

> If this is just a nice-to-have behaviour leaving it as level-2 with a WARNING should be sufficiently high.

Leaving failure reporting at level-2 does not address the [earlier](https://github.com/squid-cache/squid/pull/974#discussion_r800069683) concern, but this thread has enough information for me to be able to polish this. I will be back with specific changes shortly.

>  I do not have a preference between WARNING or NOTICE.

Whenever possible, for problem messages, we should be using one of the three standardized labels: `FATAL:`, `ERROR:`, `WARNING:`. See d816f28 for more information. That commit was about level-0/1 messages, but it is easy to adjust its rationale to become applicable to level-2 problem messages.

Reply:
> this thread has enough information for me to be able to polish this. I will be back with specific changes shortly.

Done. The primary changes are in 44c7c4d. I consider this change request addressed.